lib/internal/crypto/webidl.js

```diff
@@ -165,7 +166,16 @@ converters.object = (V, opts) => {
 };
 
 function isNonSharedArrayBuffer(V) {
-  return ObjectPrototypeIsPrototypeOf(ArrayBufferPrototype, V);
+  if (ObjectPrototypeIsPrototypeOf.call(ArrayBufferPrototype, V)) {
+    return true;
+  }
+
+  // This a fallback mechanism that handles cross-realm ArrayBuffers.
+  return (
+    typeof V === 'object' &&
+    V !== null &&
+    ObjectPrototypeToString.call(V) === '[object ArrayBuffer]'
```

Comment:
since this obeys Symbol.toStringTag, we can't rely on it; separately, the `.call` isn't needed here.